src/plans/stepsize.jl

```diff
@@ -756,6 +757,60 @@ function show(io::IO, a::NonmonotoneLinesearch)
 end
 get_message(a::NonmonotoneLinesearch) = a.message
 
+@doc raw"""
+    PolyakStepsize <: Linesearch
+
+
+Compute a step size
+
+````math
+α_i = \frac{f(p^{(k)} - f_{\text{best}}+\gamma_k}{\lVert \operatorname{grad} f(p^{(k)})} \rVert},
+````
+
+where ``f_{\text{best}}`` is the (up to now) best cost function seen until now,
+and ``\gamma_i`` is a sequence of numbers that is square summable, but not summable.
+
+# Fields
+
+* `γ`               : a function `i -> ...` representing the sequence.
+* `best_cost_value` : storing the value `f_{\text{best}}`
+
+# Constructor
+
+    PolyakStepsize(;
+        γ = i -> 1/i,
+        initial_cost_estimate=0.0
+    )
+
+initialize the Polyak stepsize to a certain sequence and an initial estimate of ``f_{\text{best}}``
+"""
+mutable struct PolyakStepsize{F,R} <: Linesearch
+    γ::F
+    best_cost_value::R
+end
+function PolyakStepsize(; γ::F=(i) -> 1 / i, initial_cost_estimate::R=0.0) where {F,R}
+    return PolyakStepsize{F,R}(γ, initial_cost_estimate)
+end
+function (ps::PolyakStepsize)(
+    amp::AbstractManoptProblem, ams::AbstractManoptSolverState, i::Int, args...; kwargs...
+)
+    # We get these by reference, so that should not allocate in general
+    M = get_manifold(amp)
+    p = get_iterate(ams)
+    X = get_gradient(amp, p)
+    # Evaluate the cost
+    c = get_cost(M, get_objective(amp), p)
+    (c < ps.best_cost_value) && (ps.best_cost_value = c)
+    α = (c - ps.best_cost_value + ps.γ(i)) / norm(M, p, X)
```

Comment:
```suggestion
    α = (c - ps.best_cost_value + ps.γ(i)) / norm(M, p, X)^2
```

Reply:
Oh then I might have to fix the code as well here.